src/components/ProjectDomain/projectdomainTable.tsx

```diff
@@ -0,0 +1,302 @@
+import React, { useState, useEffect } from "react";
+import { PackageOpen, Eye, Trash2 } from "lucide-react";
+
+import { useToast } from "@/components/ui/use-toast";
+import AddProjectDomain from "@/components/ProjectDomain/addProjectDomain";
+import { Card } from "@/components/ui/card";
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  TableRow,
+  TableHead,
+  TableCell,
+} from "@/components/ui/table";
+import {
+  Dialog,
+  DialogTrigger,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+} from "@/components/ui/dialog";
+import {
+  Tooltip,
+  TooltipTrigger,
+  TooltipContent,
+} from "@/components/ui/tooltip";
+import { axiosInstance } from "@/lib/axiosinstance";
+import { Button } from "@/components/ui/button";
+import { Switch } from "@/components/ui/switch";
+import { statusType } from "@/utils/common/enum";
+import { apiHelperService } from "@/services/projectdomain";
+
+interface DomainData {
+  _id: string;
+  label: string;
+  description: string;
+  createdAt?: string; // ISO date string
+  createdBy?: string;
+  status?: string; // User or system that created the domain
+}
+
+const ProjectDomainTable: React.FC = () => {
+  const [domainData, setDomainData] = useState<DomainData[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [noData, setNoData] = useState(false); // State to handle no data available
+  const { toast } = useToast();
+  // Function to fetch domain data
+  const fetchDomainData = async () => {
+    setLoading(true);
+    setNoData(false); // Reset noData state before fetching
+    try {
+      const response = await apiHelperService.getAllProjectdomain();
+      if (!response.data.data) {
+        setNoData(true); // Set noData if response is empty
+      } else {
+        setDomainData(response.data.data);
+      }
+    } catch (error) {
+      toast({
+        title: "Error",
+        description: "Failed to fetch domain data. Please Refresh the page.",
+        variant: "destructive", // Red error message
+      });
+
+      setNoData(true); // Handle errors by showing no data
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  // Fetch domain data on component mount
+  useEffect(() => {
+    fetchDomainData();
+  }, []);
+
+  // Handle domain deletion
+  const handleDelete = async (domainId: string) => {
+    if (!domainId) {
```

Comment:
This is a quite a wild edge case
If we don't have domainId then the domain itself should not exist by our db schema standards
If we don't have domainId then we should not be even displaying the contents in table let alone give a delete button which we know will fail 100% as we don't have domainId
Better way to handle this situation, either don't show the row in table if a mandatory field is missing OR don't give delete button in this case.
@arpit2222 can you share your thoughts on this case? What makes sense this this scenario

Reply:
it was resued by another table where there was no dynamic handling